src/param.rs

```diff
@@ -114,14 +114,15 @@ impl AudioNode for AudioParam {
 
 #[derive(Debug)]
 pub(crate) struct AudioParamProcessor {
-    value: f32,
+    value: f32, // @todo - rename to intrisic value to match the spec?
     shared_value: Arc<AtomicF64>,
     receiver: Receiver<AutomationEvent>,
     automation_rate: AutomationRate,
     default_value: f32,
     min_value: f32,
     max_value: f32,
     events: BinaryHeap<AutomationEvent>,
+    last_event: Option<AutomationEvent>, // to compute
```

Comment:
Are you sure you need the `last_event`? You're using it only for ExponentialRampToValueAtTime but I don't think it is necessary, you can set last_time and last_value equal to current time and current value, and it will give the same curve.
I may be mistaken though, let me verify this first

Reply:
Yes probably you are this is ok to store only these two values (meanwhile I also updated the linear_ramp to use them)

I will have a look at this tomorrow